src/background.js

```diff
@@ -19,3 +23,117 @@ chrome.runtime.onMessage.addListener((request) => {
     chrome.runtime.openOptionsPage();
   }
 });
+
+// Message listener from content scripts
+chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+  // do not attempt to parse the message if it is not coming from a content script
+  if (!sender.tab) {
+    return;
+  }
+
+  switch (request.operation) {
+    case "LOAD_WORKSPACES":
+      loadWorkspaces(request).then(sendResponse);
+      break;
+    case "GET_OAUTH_TOKEN":
+      getOAuthToken(request).then(sendResponse);
+      break;
+    case "REFRESH_OAUTH_TOKEN":
+      refreshOAuthToken(request).then(sendResponse);
+      break;
+
+    default:
+      console.error(`Unknown operation: ${request.operation}`);
+  }
+  return true;
+});
+
+/**
+ * Fetch workspaces from Orbit API
+ * Credentials passed as arguments since this is called from options page
+ *
+ * @param {String} accessToken
+ * @param {String} apiKey
+ *
+ * @returns {success, response, ok}
+ */
+const loadWorkspaces = async ({ accessToken, apiKey }) => {
```

Comment:
I reckon we could benefit from moving these away to separate files (ie, `requests/oauth.js` and `requests/orbit.js`?) so that this file doesn't get overly cluttered. But think that's a bit premature for just these three. We'll be adding plenty more requests to this with the full design, can do it then when we have more reason to group them